addons/mail/static/src/core/common/message_model.js

```diff
@@ -32,6 +33,23 @@ export class Message extends Record {
     attachment_ids = Record.many("ir.attachment", { inverse: "message" });
     author = Record.one("Persona");
     body = Record.attr("", { html: true });
+    richBody = Record.attr("", {
+        compute() {
+            const knownEmojisRegex = this.store.env.services["mail.core.common"].knownEmojisRegex;
+            if (!knownEmojisRegex) {
+                return this.body;
+            }
+            return markup(
+                this.body.replaceAll(knownEmojisRegex, (codepoints) => {
+                    const shortcodes =
+                        this.store.env.services["mail.core.common"].shortcodesByCodepoints[
+                            codepoints
+                        ];
+                    return `<span title="${shortcodes.join(", ")}">${codepoints}</span>`;
```

Comment:
Normally we should use format list rather than manual join for proper translation of lists.

Reply:
I would also not join the shortcodes with `,`: the whitespace is good enough, the `,` can look like part of a shortcode so this is bothering for copy-pasting.

We used plain whitespace in other places like the emoji picker search placeholder.